caddyhttp/fastcgi/fastcgi.go

```diff
@@ -121,6 +135,10 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 				resp, err = fcgiBackend.Post(env, r.Method, r.Header.Get("Content-Type"), r.Body, contentLength)
 			}
 
+			if resp.Body != nil {
```

Comment:
It would be wise to check if [resp is nil](https://github.com/abiosoft/caddy/blob/38e756b37ebe97f2cd4b7d4552bb207a43b55b97/caddyhttp/fastcgi/fcgiclient.go#L400) or to check for errors before accessing `resp`.